pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/PublishDraftVersion.java

```diff
@@ -114,46 +120,40 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
             if(canPublish) {
                 serviceResourceResolver = serviceResourceResolverProvider.getServiceResourceResolver();
             }
-            Module module = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Module.class);
+            Optional<? extends Document> document = getDocument(request, serviceResourceResolver);
```

Comment:
You shouldn't need a call to a new `getDocument()` method here, you should just be able to use the original line and just replace `Module` with `Document`.